spacialgaze-plugins/exp.js

```diff
@@ -0,0 +1,284 @@
+					});
+				});
+				//user.sendTo(room, 'You have earned ' + rewardLevel + ' for level up!');
+			} else if (level > 20) {
+				//rewardLevel = '';
+				currency = 27;
+				Economy.writeMoney(user.userid, currency, () => {
+					Economy.readMoney(user.userid, newAmount => {
+						Economy.logTransaction(Chat.escapeHTML(user.userid) + ' has received ' + currency + ' ' + (currency === 1 ? global.currencyName : global.currencyPlural) + ' from a level up.');
+					});
+				});
+				//user.sendTo(room, 'You have earned ' + rewardLevel + ' for level up!');
+			} else {
+				//rewardLevel = "-3 Stardust";
+				currency = 3;
+				Economy.writeMoney(user.userid, currency, () => {
+					Economy.readMoney(user.userid, newAmount => {
+						Economy.logTransaction(Chat.escapeHTML(user.userid) + ' has received ' + currency + ' ' + (currency === 1 ? global.currencyName : global.currencyPlural) + ' from a level up.');
+					});
+				});
+				//user.sendTo(room, 'You have earned ' + rewardLevel + ' for level up!');
+			}
+			let newLevel = SG.level(user);
+			user.sendTo(room, '|html|<center><font size=4><b><i>Level Up!</i></b></font><br />' +
+				'You have reached level ' + newLevel + '.' + /*' This will award you:<br /><b> ' + rewardLevel + */'</b></center>'
+			);
+		}
+	});
+}
+SG.addExp = addExp;
+
+function level(user) {
+	let curExp = Db.exp.get(user, 0);
+	let benchmarks = [0, 40, 90, 165, 250, 400, 600, 810, 1250, 1740, 2450, 3300, 4400, 5550, 6740, 8120, 9630, 11370, 13290, 15520, 18050, 23000, 28000, 33720, 39900, 46440, 52690, 58000, 63600, 69250, 75070, 81170, 87470, 93970, 100810, 107890, 115270, 122960, 131080, 140000];
+	for (let i = 0; i < benchmarks.length; i++) {
+		if (curExp >= 140000) return "Maxed!";
+		if (benchmarks[i] <= curExp) {
+			continue;
+		} else {
+			return i;
+		}
+	}
+}
+SG.level = level;
+
+function nextLevel(user) {
+	let curExp = Db.exp.get(user, 0);
+	let benchmarks = [0, 40, 90, 165, 250, 400, 600, 810, 1250, 1740, 2450, 3300, 4400, 5550, 6740, 8120, 9630, 11370, 13290, 15520, 18050, 23000, 28000, 33720, 39900, 46440, 52690, 58000, 63600, 69250, 75070, 81170, 87470, 93970, 100810, 107890, 115270, 122960, 131080, 140000];
+	for (let i = 0; i < benchmarks.length; i++) {
+		if (curExp >= 140000) return "no more level ups :(";
+		if (benchmarks[i] <= curExp) {
+			continue;
+		} else {
+			return benchmarks[i] - curExp + " exp";
+		}
+	}
+}
+SG.nextLevel = nextLevel;
+
+exports.commands = {
+	level: 'exp',
+	xp: 'exp',
+	exp: function (target, room, user) {
+		if (!this.runBroadcast()) return;
+		if (!target) target = user.name;
+
+		const targetId = toId(target);
+
+		EXP.readExp(targetId, exp => {
+			this.sendReplyBox('<b>' + SG.nameColor(targetId, true) + '</b> has ' + exp + ' exp and is level ' + SG.level(targetId) + ' and has ' + SG.nextLevel(targetId) + ' until the next level.');
```

Comment:
...`and needs ' + SG.nextLevel(targetId) + ' to reach the next level.`

That part of the current message sounds clunky.